corporate/lib/stripe.py

```diff
@@ -651,10 +651,19 @@ def downgrade_now_without_creating_additional_invoices(realm: Realm) -> None:
     plan.next_invoice_date = next_invoice_date(plan)
     plan.save(update_fields=["invoiced_through", "next_invoice_date"])
 
-def downgrade_at_the_end_of_billing_cycle(realm: Realm) -> None:
+    event_time = timezone_now()
+    RealmAuditLog.objects.create(realm=realm, acting_user=acting_user,
+                                 event_time=event_time,
+                                 event_type=RealmAuditLog.REALM_DOWNGRADED_PLAN_WITHOUT_CREATING_ADDITIONAL_INVOICES)
+
+def downgrade_at_the_end_of_billing_cycle(realm: Realm, acting_user: Optional[UserProfile]=None) -> None:
     plan = get_current_plan_by_realm(realm)
     assert(plan is not None)
     do_change_plan_status(plan, CustomerPlan.DOWNGRADE_AT_END_OF_CYCLE)
+    event_time = timezone_now()
+    RealmAuditLog.objects.create(realm=realm, acting_user=acting_user,
+                                 event_time=event_time,
+                                 event_type=RealmAuditLog.REALM_DOWNGRADED_PLAN_AT_END_OF_BILLING_CYCLE)
```

Comment:
Actually let's skip the commit for now. I want to think a bit more on the type of logging here.